packages/python/accern_data/accern_data.py

```diff
@@ -580,68 +646,87 @@ def download_range(
         if output_path is None:
             output_path = "./"
         os.makedirs(output_path, exist_ok=True)
-        expected_records: List[int] = []
-        if mode is None:
-            valid_mode = self.get_mode()
-        elif isinstance(mode, Mode):
-            valid_mode = mode.get_instance()
-        elif isinstance(mode, str):
-            valid_mode = self.parse_mode(mode)
-        else:
-            valid_mode = self.parse_mode(*mode)
-        if filters is None:
-            valid_filters = self.get_filters()
-        else:
-            valid_filters = self.parse_filters(
-                {**self.get_filters(), **self.validate_filters(filters)})
+        valid_mode = self._get_valid_mode(mode, deep_copy=True)
```

Comment:
just pass the args to iterate_range

Reply:
we need `filters` for `fetching_info` & `mode` for writing files.

Reply:
see my other comment

Reply:
also why do you need the other two for writing? that is all handled by mode